pkg/golinters/revive.go

```diff
@@ -0,0 +1,187 @@
+package golinters
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/token"
+	"io/ioutil"
+	"strings"
+
+	"github.com/mgechev/dots"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
+	"github.com/golangci/golangci-lint/pkg/result"
+
+	reviveConfig "github.com/mgechev/revive/config"
+	"github.com/mgechev/revive/lint"
+	"golang.org/x/tools/go/analysis"
+)
+
+const (
+	reviveName = "revive"
+)
+
+// jsonObject defines a JSON object of an failure
+type jsonObject struct {
+	Severity     lint.Severity
+	lint.Failure `json:",inline"`
+}
+
+// NewNewRevive returns a new Revive linter.
+func NewRevive(cfg *config.ReviveSettings) *goanalysis.Linter {
+	var (
+		issues   []goanalysis.Issue
+		analyzer = &analysis.Analyzer{
+			Name: goanalysis.TheOnlyAnalyzerName,
+			Doc:  goanalysis.TheOnlyanalyzerDoc,
+		}
+	)
+
+	return goanalysis.NewLinter(
+		reviveName,
+		"Fast, configurable, extensible, flexible, and beautiful linter for Go",
+		[]*analysis.Analyzer{analyzer},
+		nil,
+	).WithContextSetter(func(lintCtx *linter.Context) {
+		analyzer.Run = func(pass *analysis.Pass) (interface{}, error) {
+			var (
+				files = []string{}
+			)
+
+			for _, file := range pass.Files {
+				files = append(files, pass.Fset.PositionFor(file.Pos(), false).Filename)
+			}
+
+			conf, err := SetReviveConfig(cfg)
+			if err != nil {
+				return nil, err
+			}
+
+			formatter, err := reviveConfig.GetFormatter("json")
+			if err != nil {
+				return nil, err
+			}
+
+			revive := lint.New(ioutil.ReadFile)
+
+			lintingRules, err := reviveConfig.GetLintingRules(conf)
+			if err != nil {
+				return nil, err
+			}
+
+			packages, err := dots.ResolvePackages(files, normalizeSplit([]string{}))
```

Comment:
`normalizeSplit([]string{})` seems useless here, because it will always return an empty slice.

Reply:
Good catch, this string slice should have contained a list of excluded packages. I decided to remove "normalizeSplit" and just pass an empty string slice because it would make more sense to just use golanci-lint's option to skip certain directories `--skip-dirs` then add another exclude setting. Hopefully that sounds good?